doc/guides/compartments.rst

```diff
@@ -0,0 +1,295 @@
+
+.. code-block:: python
+
+    def sodium_pump(cytoplasm, extracellularSpace):
+        ...
+        return "Update: Decrease ATP concentration in cytoplasm by x mM"
+
+A function's modularity comes from the fact that we can pass in different
+objects for the ``cytoplasm`` parameter, even objects the function
+authors hadn't thought of. ``cytoplasm`` is like the port, to which we
+can provide any store we like.
+
+How Processes Define Ports
+==========================
+
+A process specifies its port names in its ``ports_schema`` function.
+For example, the :py:class:`vivarium.processes.tree_mass.TreeMass`
+schema is created like this:
+
+.. code-block:: python
+
+    def ports_schema(self):
+        return {
+            'global': {
+                'initial_mass': {
+                    '_default': self.parameters['initial_mass'],
+                    '_updater': 'set',
+                    '_divider': 'split',
+                },
+                'mass': {
+                    '_default': self.parameters['initial_mass'],
+                    '_emit': True,
+                    '_updater': 'set',
+                    '_divider': 'split',
+                },
+            },
+        }
+
+The top level keys are the port names. In this case, the only port is
+``global``. The next level of keys define the variables expected to be
+in each port. Here, we expect ``global`` to have the variables
+``initial_mass`` and ``mass``.
+
+When the process is asked to provide an update to the model state, it is
+only provided the variables it specifies. For example, it might get a
+model state like this:
+
+.. code-block:: python
+
+    {
+        'global': {
+            'initial_mass': 0,
+            'mass': 1339,
+        },
+    }
+
+This would happen even if the store linked to the ``global`` port
+contained more variables. We call this stripping-out of variables the
+process doesn't need :term:`masking`.
+
+----------
+Topologies
+----------
+
+How do we specify which store goes with which port? To continue the
+function analogy from above, we need something analogous to this:
+
+.. code-block:: python
+
+    cell = Cell()
+    bloodVessel = BloodVessel()
+    # We need something like the line below
+    update = sodium_pump(cytoplasm=cell, extracellularSpace=bloodVessel)
+
+When we call ``sodium_pump``, we specify which objects go with which
+parameters. Analogously, we specify the mapping between ports and stores
+using a :term:`topology`.
+
+Defining Topologies
+===================
+
+We define topologies as dictionaries with process names as keys and
+dictionaries (termed "sub-dictionaries") as values. These
+sub-dictionaries have port names as keys and store names as values. For
+example, the topology for the ATP example we have been considering might
+look like this:
+
+.. code-block:: python
+
+    {
+        'sodium_pump': {
+            'cytoplasm': 'cell',
+            'extracellularSpace': 'bloodVessel',
```

Comment:
we now link to stores in tuples to specify their relative compartment: ```('cell',)```. Might be worth giving an example of going up one level: ```('..', 'boundary')```

Edit -- you describe this below already. That looks sufficient.

Reply:
I'll fix this part to use paths to be consistent. I also describe paths in more detail in the hierarchy guide